tools/sof-dump-status.py

```diff
@@ -2,10 +2,20 @@
 
 import subprocess
 import os
+import sys
+
 from common import format_pipeline, export_pipeline
 
+
+class NoMapException(Exception):
+    """Raise if some element doesn't have an associated map."""
```

Comment:
Overkill, see below (plus this looks like a pure duplicate of Python's built-in `KeyError` with no added value)

Reply:
This isn't what I was trying to go for. `NoMapException` is meant to be used when you're trying to map some element to another using multiple maps.
Right now, I don't think that idea is very clear because imx_soc_id_to_dsp_name is the only map between a soc_id and a dsp_name but if more possible maps between soc_id and dsp_name were to be introduced I believe it would make more sense. Example:

```
            if self.dt_info["soc_id"] in self.imx_soc_id_to_dsp_name:
                dsp_name = self.imx_soc_id_to_dsp_name[self.dt_info["soc_id"]]
            elif self.dt_info["soc_id"] in self.platform1_soc_id_to_dsp_name:
                dsp_name = ...
            elif self.dt_info["soc_id"] in self.platform2_soc_id_to_dsp_name:
                dsp_name = ...
            elif self.dt_info["soc_id"] in self.platform3_soc_id_to_dsp_name:
                dsp_name = ...
            else:
                raise NoMapException("Current DT-based platform doesn't have a soc_id to dsp_name map.")
```

I believe that this:

```
            if self.dt_info["soc_id"] in self.imx_soc_id_to_dsp_name:
                dsp_name = self.imx_soc_id_to_dsp_name[self.dt_info["soc_id"]]
            else:
                raise NoMapException("Current DT-based platform doesn't have a soc_id to dsp_name map.")
```

could in fact be turned into this:

```
            dsp_name = self.imx_soc_id_to_dsp_name[self.dt_info["soc_id"]]
```
and if there's no i.MX mapping just let it raise `KeyError` but if more maps were to be introduced I believe the first example would be more suited.

Reply:
I'm still not convinced sorry.

```python
all_dsp_names = { **platform1_soc_id_to_dsp_name, **platform2_soc_id_to_dsp_name, ...
dsp_name = all_dsp_names[dt_info["soc_id"]]
```

2 lines, done! Less is more.

Reply:
Thanks ! This is so much better than what I was trying to go for.